### PR TITLE
Add timestamps flag to show PipelineRun log with timestamps

### LINKED ISSUE #1749: Add 'timestamps' flag to show logs with timestamps
### Feature request
Add `timestamps` flag to show logs with timestamps in the same as `kubectl`

### Use case
It's useful when the `Task`'s logs don't have timestamps.

### UI Example
```
tkn pipelinerun logs --timestamps
```

### PR DESCRIPTION
**Pull request by seongpyoHong**
# Changes
Resolves #1749 
Add `timestamps` flag to show PipelineRun log with timestamps.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
  - I don't know how to test this PR in `container_test.go`
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes
```release-note
Add timestamps flag to show PipelineRun log with timestamps
```

### CODE REVIEW

**Comment by chmouel** [comment]
this is great thanks, but can we get a unit test for this?

**Comment by seongpyoHong** [comment]
Thanks, @chmouel for your feedback.

I also think unit test need. But, it is difficult to write a unit test. So, I'd like to ask your opinion.

In unit test(may be `pkg/pods/container_test.go`), it compares the predefined test data with the return value of the `containerLogs`  function.

There are some problems to execute unit test.  
1. When adding timestamp to the log of test data: 
It is not possible to distinguish whether it is a log added by application  or added due to the timestamp flag. (when timestamps flag is true, logs return with the timestamp)
2. Add a separate `Timestamps` field to the test data: we can't this field in the containerLogs function.
  
(Perhaps for similar reasons, there is no unit test for timestamps in [kubectl](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/logs/logs_test.go))

File: pkg/cmd/pipelinerun/logs.go [hunk 1]
```diff
@@ -80,10 +80,10 @@ Show the logs of PipelineRun named 'microservice-1' for all Tasks and steps (inc
 	c.Flags().BoolVarP(&opts.Last, "last", "L", false, "show logs for last PipelineRun")
 	c.Flags().BoolVarP(&opts.Fzf, "fzf", "F", false, "use fzf to select a PipelineRun")
 	c.Flags().BoolVarP(&opts.Follow, "follow", "f", false, "stream live logs")
+	c.Flags().BoolVarP(&opts.Timestamps, "timestamps", "", false, "show logs with timestamp")
```
Comment (vinamra28) [inline pkg/cmd/pipelinerun/logs.go:83]: should we rename the flag to probably `--show-timestamps` or something of that sort?
cc @piyush-garg @chmouel @pradeepitm12 @vdemeester

**Comment by chmouel** [comment]
> (Perhaps for similar reasons, there is no unit test for timestamps in kubectl)

fair enough :) 

thanks for looking at it

/lgtm

File: pkg/cmd/pipelinerun/logs.go (hunk 1 above)
Comment (vinamra28) [reply pkg/cmd/pipelinerun/logs.go:83]: makes sense. Thanks Chmouel 😄

**Review by vinamra28 — COMMENTED** [review]
PR looks good. Can we get this done for TaskRun as well?
/lgtm

**Review by vdemeester — APPROVED** [review]

**Comment by seongpyoHong** [comment]
> Can we get this done for TaskRun as well?

Yes, it can apply for TaskRun and also all resources which have `logs` command. (such as TaskRun, ClusterTask, Task, Pipeline, EventListener).

How about add `timestamps` command for all resources which has `logs` command?

**Comment by chmouel** [comment]
> which is better to add timestamps command for all resources which have logs command in this PR or to separate PR for each resource?

as you like but I'd rather review only one PR than multiple and less random CI failures too 🙃

**Comment by seongpyoHong** [comment]
> as you like but I'd rather review only one PR than multiple and less random CI failures too 🙃

Ok, I just add flag for only PipelineRun in this PR.

**Pull request merged**